tests/jballerina-unit-test/src/test/resources/test-src/object/object_functions_with_default_parameters_negative.bal

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public class Class1 {
+    int i = 0;
+
+    public function foo(int arg1 = i) { // error
```

Comment:
Shall we add more tests to include `self.i` also??

Reply:
Added